neuland/geobase/create_neuland_geo.C

```diff
@@ -0,0 +1,76 @@
+/******************************************************************************
+ *   Copyright (C) 2023 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2023 Members of R3B Collaboration                          *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+#include "./helper_neuland_geometry.C"
+#include "TGeoManager.h"
+#include "TMath.h"
+#include <iomanip>
+#include <iostream>
+
+void create_neuland_geo(const Int_t nPlanes = 26, const Double_t distance = 1136.105, const TString geoTag = "v2023.3")
+{
+
+    new FairGeoLoader("TGeo", "FairGeoLoader");
+    gGeoManager->SetName("NEULANDgeom");
+
+    TGeoVolume* volPaddle = BuildPaddleVolume();
+    TGeoRotation* rot0 = nullptr;
+    TGeoRotation* rot90 = new TGeoRotation();
+    rot90->RotateZ(90.);
+    TGeoRotation* rot45 = new TGeoRotation();
+    rot45->RotateY(45.);
+
+    TGeoVolume* volNeuland = new TGeoVolumeAssembly("volNeuland");
+    Int_t nindex = 0, nPlane = 0;
+    for (Double_t b = -nPlanes * gPaddleDistance + gPaddleDistance; b < nPlanes * gPaddleDistance;
+         b += gPaddleDistance * 2)
+    {
+        nPlane++;
+        for (Double_t a = -gPaddlesPerPlane * gPaddleDistance + gPaddleDistance; a < gPaddleDistance * gPaddlesPerPlane;
+             a += gPaddleDistance * 2)
+        {
+            nindex++;
+            if (nPlane % 2 == 1)
+            {
+                volNeuland->AddNode(volPaddle, nindex, new TGeoTranslation(0, a, b));
+            }
+            else
+            {
+                volNeuland->AddNode(volPaddle, nindex, new TGeoCombiTrans(a, 0, b, rot90));
+            }
+        }
+    }
+
+    TGeoVolume* top = new TGeoVolumeAssembly("TOP");
+    gGeoManager->SetTopVolume(top);
+    top->AddNode(volNeuland, 1, new TGeoCombiTrans(0., 0., distance + nPlanes + gPaddleDistance, rot0));
```

Comment:
Yes, we do this with in-beam detectors because they could be moved during the beam time. But this is not the case for NeuLAND, right? Did you move it during the last 4-6 experiments?

From my side there is not problem with setting its position in (0,0,0), but in this case all the users must have very clear the position of this detector to avoid misunderstandings. Maybe NeuLAND experts (@igasparic) can comment on what they want to have here.

Reply:
Even if we will never move it, it's still kind of bad to define the distance value here, for the following reasons:
1. Distance value can also be set through R3BNeuland constructor. One quantity should only be defined through one method. If the value can be set here and there, it's bound to cause trouble in the future.
2. The geometry file can also be used in the simulation, which could use different distances.
3. Geometry file should only contains geometrical information. We don't need to create different geo files for the exactly same geo configurations of Neuland with different distances., like neuland_geo_13dp_1600cm, neuland_geo_13dp_1500cm, neuland_geo_13dp_1400cm ...

Reply:
Yes, for me it is also better to have something general for all the detectors and passive elements like GLAD.

Reply:
I agree with Yanzhao. I would leave distance to a user to decide. For instance when a new double plane is added then we usually move other planes as well to pack them tightly. So although the holding frame is fixed the detector position with respect to the cave changes.

Reply:
Done!

Reply:
Please, add to the R3B-Wiki the position of NeuLAND for each experiment, so users can take the right values for simulations, analysis, etc.

Reply:
I would say remove the distance variable in the MACRO. If someone in the future see that he can define a distance in the macro, he will either probably define the same distance value again or be puzzled why there are two ways to set the distance value.